### ISSUE TITLE
Custom config not working

### ISSUE THREAD

**Issue by cookie10codes** [percipioglobal/craft-colour-swatches#87]
The custom config to add colours in the settings doesn't work

<img width="792" alt="Screenshot 2022-04-27 at 10 11 44" src="https://user-images.githubusercontent.com/20947573/165472576-321384e9-d0d6-4b43-9034-fee304bb6460.png">

**Comment by Antimated** [comment]
Just installed colour swatches version 4.0.0 on craft version 4.0.0.1 and I'm not able to use the plugin. Custom and non-custom color settings do not work.

**My scenario**

1. Have a matrix
2. Add a colour swatch field
3. Select a colour config (note: whenever I pick a config no previews are viewed initially, only after save)
4. Go to a page that uses this matrix

I get the following error

```
An Error occurred while handling another error:
TypeError: percipiolondon\colourswatches\models\ColourSwatches::__construct(): Argument #1 ($value) must be of type string, null given, called in /Users/nicholas/Sites/msk/vendor/percipioglobal/craft-colour-swatches/src/fields/ColourSwatches.php on line 109 and defined in /Users/nicholas/Sites/msk/vendor/percipioglobal/craft-colour-swatches/src/models/ColourSwatches.php:31
Stack trace:
#0 /Users/nicholas/Sites/msk/vendor/percipioglobal/craft-colour-swatches/src/fields/ColourSwatches.php(109): percipiolondon\colourswatches\models\ColourSwatches->__construct(NULL)
#1 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/base/Element.php(4661): percipiolondon\colourswatches\fields\ColourSwatches->normalizeValue(NULL, Object(craft\elements\MatrixBlock))
#2 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/base/Element.php(3653): craft\base\Element->normalizeFieldValue('accent')
#3 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(238): craft\base\Element->getFieldValue('accent')
#4 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\MatrixBlock), false)
#5 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\MatrixBlock), false)
#6 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#7 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#8 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\MatrixBlock), false)
#9 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#10 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#11 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fields/Matrix.php(1192): craft\models\FieldLayout->createForm(Object(craft\elements\MatrixBlock))
#12 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fields/Matrix.php(692): craft\fields\Matrix->_getBlockTypeInfoForInput(Object(craft\elements\Entry), Array, 'olyrwplgwx')
#13 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/base/Field.php(453): craft\fields\Matrix->inputHtml(Array, Object(craft\elements\Entry))
#14 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(257): craft\base\Field->getInputHtml(Object(craft\elements\db\MatrixBlockQuery), Object(craft\elements\Entry))
#15 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\Entry), false)
#16 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\Entry), false)
#17 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#18 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#19 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\Entry), false)
#20 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#21 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#22 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/controllers/ElementsController.php(717): craft\models\FieldLayout->createForm(Object(craft\elements\Entry), false, Array)
#23 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/controllers/ElementsController.php(426): craft\controllers\ElementsController->_prepareEditor(Object(craft\elements\Entry), true, Object(craft\web\Response), 'main-form', Object(Closure), Object(Closure), Object(Closure))
#24 [internal function]: craft\controllers\ElementsController->craft\controllers\{closure}(Object(craft\web\Response), 'main-form')
#25 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(108): call_user_func(Object(Closure), Object(craft\web\Response), 'main-form')
#26 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(48): craft\web\CpScreenResponseFormatter->_formatTemplate(Object(craft\web\Response), Object(craft\web\CpScreenResponseBehavior))
#27 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\CpScreenResponseFormatter->format(Object(craft\web\Response))
#28 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/Response.php(275): yii\web\Response->prepare()
#29 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#30 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/web/ErrorHandler.php(136): yii\web\Response->send()
#31 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/ErrorHandler.php(185): yii\web\ErrorHandler->renderException(Object(TypeError))
#32 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/base/ErrorHandler.php(135): craft\web\ErrorHandler->renderException(Object(TypeError))
#33 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/ErrorHandler.php(64): yii\base\ErrorHandler->handleException(Object(TypeError))
#34 [internal function]: craft\web\ErrorHandler->handleException(Object(TypeError))
#35 {main}
Previous exception:
TypeError: percipiolondon\colourswatches\models\ColourSwatches::__construct(): Argument #1 ($value) must be of type string, null given, called in /Users/nicholas/Sites/msk/vendor/percipioglobal/craft-colour-swatches/src/fields/ColourSwatches.php on line 109 and defined in /Users/nicholas/Sites/msk/vendor/percipioglobal/craft-colour-swatches/src/models/ColourSwatches.php:31
Stack trace:
#0 /Users/nicholas/Sites/msk/vendor/percipioglobal/craft-colour-swatches/src/fields/ColourSwatches.php(109): percipiolondon\colourswatches\models\ColourSwatches->__construct(NULL)
#1 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/base/Element.php(4661): percipiolondon\colourswatches\fields\ColourSwatches->normalizeValue(NULL, Object(craft\elements\MatrixBlock))
#2 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/base/Element.php(3653): craft\base\Element->normalizeFieldValue('accent')
#3 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(238): craft\base\Element->getFieldValue('accent')
#4 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\MatrixBlock), false)
#5 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\MatrixBlock), false)
#6 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#7 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#8 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\MatrixBlock), false)
#9 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#10 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#11 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fields/Matrix.php(1192): craft\models\FieldLayout->createForm(Object(craft\elements\MatrixBlock))
#12 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fields/Matrix.php(692): craft\fields\Matrix->_getBlockTypeInfoForInput(Object(craft\elements\Entry), Array, 'uzjdmscbhb')
#13 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/base/Field.php(453): craft\fields\Matrix->inputHtml(Array, Object(craft\elements\Entry))
#14 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(257): craft\base\Field->getInputHtml(Object(craft\elements\db\MatrixBlockQuery), Object(craft\elements\Entry))
#15 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\Entry), false)
#16 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\Entry), false)
#17 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#18 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#19 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\Entry), false)
#20 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#21 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#22 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/controllers/ElementsController.php(717): craft\models\FieldLayout->createForm(Object(craft\elements\Entry), false, Array)
#23 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/controllers/ElementsController.php(426): craft\controllers\ElementsController->_prepareEditor(Object(craft\elements\Entry), true, Object(craft\web\Response), 'main-form', Object(Closure), Object(Closure), Object(Closure))
#24 [internal function]: craft\controllers\ElementsController->craft\controllers\{closure}(Object(craft\web\Response), 'main-form')
#25 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(108): call_user_func(Object(Closure), Object(craft\web\Response), 'main-form')
#26 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(48): craft\web\CpScreenResponseFormatter->_formatTemplate(Object(craft\web\Response), Object(craft\web\CpScreenResponseBehavior))
#27 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\CpScreenResponseFormatter->format(Object(craft\web\Response))
#28 /Users/nicholas/Sites/msk/vendor/craftcms/cms/src/web/Response.php(275): yii\web\Response->prepare()
#29 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#30 /Users/nicholas/Sites/msk/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#31 /Users/nicholas/Sites/msk/web/index.php(26): yii\base\Application->run()
#32 /Users/nicholas/.composer/vendor/laravel/valet/server.php(234): require('/Users/nicholas...')
#33 {main}
```

**Comment by michtio** [comment]
Thank you for the report @Antimated 
Could you also share your colours config? 

We will test this out tomorrow 👍🏻

**Comment by Antimated** [comment]
@michtio  here you go!
```
<?php

return [

    // Custom  palettes, fixed options [label, default (boolean), colour (array(colour, customOptions)) ]
    'palettes' => [
        'Accent theme' => [  // custom label (required)
            [
                'label'   => 'Zwarte achtergrond, witte navigatie/logo',
                'default' => false,
                'color'   =>  [
                    [
                        'color'             => '#000000',              
                        'background'        => 'bg-black',
                    ],
                    [
                        'color'             => '#ffffff',               
                        'text'              => 'text-white',
                    ]
                ]
            ],
            [
                'label'   => 'Witte achtergrond, zwarte navigatie/logo',
                'default' => false,
                'color'   =>  [
                    [
                        'color'             => '#ffffff',              
                        'background'        => 'bg-white',
                    ],
                    [
                        'color'             => '#000000',               
                        'text'              => 'text-black',
                    ]
                ]
            ],
            [
                'label'   => 'Rode achtergrond, witte navigatie/logo',
                'default' => false,
                'color'   =>  [
                    [
                        'color'             => '#FF6650',              
                        'background'        => 'bg-accent-red',
                    ],
                    [
                        'color'             => '#ffffff',               
                        'text'              => 'text-white',
                    ]
                ]
            ],
        ]
    ]

];
```

**Comment by michtio** [comment]
Thank you @Antimated ! 

Going to come up with a fix ASAP

**Comment by michtio** [comment]
@Antimated the issue has now been resolved in `4.0.1` our apologies for the inconvenience
